src/common/snippets/include/snippets/lowered/loop_info.hpp

```diff
@@ -141,6 +134,11 @@ class LoopInfo {
         std::for_each(m_output_ports.cbegin(), m_output_ports.cend(), caller);
     }
 
+    /**
+     * @brief Reorder ALL Loop Ports by execution order of expressions
+     */
+    virtual void reorder_ports() = 0;
```

Comment:
Honestly, I wanted to remove it also! And I fully agree with you. Moreover, firstly I moved it into private section.
However, I found the place when we have to call `sort_loops`: for example, you have outer and inner loops. When you move expressions in inner loop which are LoopPorts of the both loops -> you should sort ports of the both Loops!
Otherwise, outer loop will have incorrect order of ports.
For example,
- [FuseLoops](https://github.com/openvinotoolkit/openvino/blob/81d9688bfc653bcb353521d539cfd5e55047c4b1/src/common/snippets/src/lowered/pass/fuse_loops.cpp#L94): when you move expression with inner Loop in outer Loop
- [ExtractLoopInvariants](https://github.com/openvinotoolkit/openvino/blob/81d9688bfc653bcb353521d539cfd5e55047c4b1/src/common/snippets/src/lowered/pass/extract_loop_invariants.cpp#L163): when you move expression from innermost Loop and change loop port order of outer loop

To automatically sort LoopPorts after each LinearIR changes I suggest to call `loop_manager->sort_loop_ports()` in each `insert` or `move` of LinearIR: https://github.com/openvinotoolkit/openvino/pull/25623#discussion_r1682666609
But I'm not sure that it's good idea since it's we will introduce Loop-mechanic to LinearIR base methods.

Reply:
It seems that we need to sort loop ports only if we move/insert expressions that are loop ports.
1) So one solution is that LIR should inform loop_manager that it's moving expression X, so the manager would trigger resort of a proper loop if X is its port. It's likely that we will have to store all loop ports in a hash table for faster checks.
2) Another option is to store `LoopPorts` in a map instead of [a vector](https://github.com/a-sidorova/openvino/blob/8edd5bc4609cfad3b752228aa22bbae3d95a69a9/src/common/snippets/include/snippets/lowered/loop_info.hpp#L189-L190) with `m_exec_num` as keys. Then all insertions and loop port manipulations would maintain the sorted order. The only event that would require rebuilding of this map is expression renumeration which won't happen too often (probably almost never) after this PR. In this case again, LIR should inform loop_manager that it renumerated expressions, so the manager would ask all loop infos to sort their port maps (i.e. create maps with the same values but new keys).

Anyways, let's not do it in this PR (dynamism is more important). Could you please create a ticket with the problem description and possible solutions + reference it in the places where we explicitly call port sorting.  Thank you 🙂 

I think we'll need to address this issue once and for all.